src/dataflow/src/storage/render/debezium.rs

```diff
@@ -118,6 +121,304 @@ where
+                                                1,
+                                            ));
+                                            continue;
+                                        }
+                                    }
+                                }
+                                None => true,
+                            };
+
+                            match (should_use, tx_id) {
+                                (true, Some((tx_id, tx_time))) => {
+                                    let tx_cap_entry = match tx_cap_map.entry(tx_id.clone()) {
+                                        Entry::Occupied(e) => e,
+                                        Entry::Vacant(_) => panic!("Must have cap if using record"),
+                                    };
+
+                                    let mut session = output.session(tx_cap_entry.get());
+                                    match value.iter().nth(before_idx).unwrap() {
+                                        Datum::List(l) => {
+                                            session.give((Ok(Row::pack(&l)), tx_time, -1));
+                                        }
+                                        Datum::Null => {}
+                                        d => {
+                                            panic!("type error: expected record, found {:?}", d)
+                                        }
+                                    }
+                                    match value.iter().nth(after_idx).unwrap() {
+                                        Datum::List(l) => {
+                                            session.give((Ok(Row::pack(&l)), tx_time, 1));
+                                        }
+                                        Datum::Null => {}
+                                        d => {
+                                            panic!("type error: expected record, found {:?}", d)
+                                        }
+                                    }
+                                    match tx_event_count.entry(tx_id.clone()) {
+                                        Entry::Occupied(mut e) => {
+                                            let count = e.get_mut();
+                                            *count -= 1;
+                                            if *count == 0 {
+                                                // Must drop the capability to allow the output frontier to progress
+                                                let _ = tx_cap_entry.remove_entry();
+                                                let _ = e.remove_entry();
+                                            }
+                                        }
+                                        Entry::Vacant(_) => panic!("need event count for tx_id {:?}", tx_id),
+                                    }
+                                },
+                                (true, None) => {
```

Comment:
Hm, shouldn't this be an error or ignored? In which case do we care about data that cannot be matched to a transaction?